.github/workflows/industrial_ci_action.yml

```diff
@@ -0,0 +1,32 @@
+# This config uses industrial_ci (https://github.com/ros-industrial/industrial_ci.git).
+# For troubleshooting, see readme (https://github.com/ros-industrial/industrial_ci/blob/master/README.rst)
+
+name: BuildAndTest
+
+on:
+  push:
+    branches: [ ros2 ]
+  pull_request:
+    branches: [ ros2 ]
+
+jobs:
+  industrial_ci:
+    strategy:
+      matrix:
+        env:
+          - {ROS_DISTRO: foxy, ROS_REPO: main, CLANG_TIDY: pedantic}
+          - {ROS_DISTRO: foxy, ROS_REPO: testing}
+    env:
+      UPSTREAM_WORKSPACE: geometric_shapes.repos
+      CCACHE_DIR: ~/.ccache
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ccache-${{ matrix.env.ROS_DISTRO }}-${{ matrix.env.ROS_REPO }}-${{ github.sha }}  
```

Comment:
I see the following warning for the 3rd job:
```
Post job cleanup.
Unable to reserve cache with key ccache-foxy-main-190fd59788645cbe4195b0056d862e1f21363f2d, another job may be creating this cache.
```
I think this is due to the fact that the 1st and 3rd build jobs try to create the same cache key: The key doesn't distinguish between CLANG_TIDY on/off. Should be resolved with my suggestion above. In general, you need to take care considering all build-relevant env variables.